### PR TITLE
JERupdate

### PR DESCRIPTION
**Pull request by cihar29**
This pull request encompasses all of the updates made by Chad Harrington for the calculation of the MC truth JER.

### CODE REVIEW

File: JetAnalyzers/bin/JERWriter.h [hunk 1]
```diff
@@ -22,8 +22,10 @@
 #include <vector>
 #include <map>
 #include <set>
+#include <math.h>
 
-
+#include "TFile.h"
+#include "TProfile.h"
```
Comment (aperloff) [inline JetAnalyzers/bin/JERWriter.h]: Move lines 27-28 right below the TF1.h library and change the TF1.h library from having <> around it to "".

File: JetAnalyzers/bin/JERWriter.h [hunk 2]
```diff
@@ -57,211 +63,323 @@ struct JERWriter
   JERWriter(const std::string& n,
 	    const std::string& e,
 	    const std::string& p,
-	    const bool b)
-    : alg(n),era(e),prefix(p),isptres(b){init();}
+	    const bool b,
+            const bool b2,
+	    const bool b3,
+        std::string s)
```
Comment (aperloff) [inline JetAnalyzers/bin/JERWriter.h]: Can we move more of these to the same line? Having them each on individual lines is silly. Just don't allow them to go beyond 120 characters per line.

**Comment by aperloff** [comment]
Please fix the spacing. It is making commenting this pull request quite a hassle as the various lines won't line up.

File: JetAnalyzers/python/JetCorrection_cff.py [hunk 3]
```diff
@@ -126,7 +126,8 @@
 #
 # PFchs JEC PRODUCERS NOT DEFINED IN DEFAULTJEC_CFF
 #
-ak5PFL1Offset = ak4CaloL1Offset.clone(algorithm = 'AK5PF')
+
```
Comment (aperloff) [inline JetAnalyzers/python/JetCorrection_cff.py]: Why is there an extra line here?

File: JetAnalyzers/python/JetCorrection_cff.py [hunk 4]
```diff
@@ -339,42 +350,42 @@
...
-    era         = cms.string('Summer11'),
+    #era         = cms.string('Summer11'),
     level       = cms.string('L1FastJet'),
     algorithm   = cms.string('AK1PUPPI'),
-    section     = cms.string(''),
-    srcRho      = cms.InputTag('kt6PFJets','rho'),
-    useCondDB = cms.untracked.bool(True)
+    #section     = cms.string(''),
+    srcRho      = cms.InputTag('fixedGridRhoFastjetAll'),
+    #useCondDB = cms.untracked.bool(True)
```
Comment (aperloff) [inline JetAnalyzers/python/JetCorrection_cff.py]: Why did you comment out certain pieces of this ESProducer?

File: JetAnalyzers/python/JetCorrection_cff.py [hunk 5]
```diff
@@ -339,42 +350,42 @@
...
     )
 ak2PUPPIL1Fastjet  = ak1PUPPIL1Fastjet.clone( algorithm = 'AK2PUPPI' )
 ak3PUPPIL1Fastjet  = ak1PUPPIL1Fastjet.clone( algorithm = 'AK3PUPPI' )
-ak4PUPPIL1Fastjet  = ak1PUPPIL1Fastjet.clone( algorithm = 'AK4PUPPI' )
+ak4PUPPIL1Fastjet  = ak1PUPPIL1Fastjet.clone( algorithm = 'AK4PFPuppi' )
```
Comment (aperloff) [inline JetAnalyzers/python/JetCorrection_cff.py]: Why not do PUPPI-->PFPuppi for all cone sizes?

**Comment by cihar29** [comment]
Bro, I've updated the files in answer to all of the previous questions.

**Comment by aperloff** [comment]
Okay, everything seems in order. I'm going to merge this pull request.

**Pull request merged**